### PR TITLE
EI-342 Add host twin helpers to high level API

### PR DESCRIPTION
**Pull request by timchurchard**
- Add DelegateControlByPrivateExponent and GetOwnershipOfTwinFromPrivateExponent helper functions to highlevelApi to ease working with private key hex.

### CODE REVIEW

File: pkg/api/highLevelApi_test.go [hunk 1]
```diff
@@ -34,6 +38,12 @@ func TestCreateTwinWithControlDelegation(t *testing.T) {
 	assert.Assert(t, twinID.Did() == "did:iotics:iotXqziDRUAqcb7Sd5NcCKMHi2DCvnNKmY7B")
 	assert.Assert(t, resolver.CountDiscover.Value() == 2) // override false causes a resolver discover
 	assert.Assert(t, resolver.CountRegister.Value() == 2) // create + delegate with 1 register call
+
+	doc, err := resolver.GetDocument(twinID.Did())
+	assert.NilError(t, err)
+	assert.Check(t, len(doc.DelegateControl) == 1)
+	assert.Check(t, doc.DelegateControl[0].ID == "#Delegation")
```
Comment (vtermanis) [inline pkg/api/highLevelApi_test.go:45]: (unimportant) could check against `opts.DelegationName` instead?

File: pkg/api/hosttwin.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package api
```
Comment (vtermanis) [inline pkg/api/hosttwin.go:1]: I appreciate that these methods (and tests) are there to enable the host twin functionality. But from my understanding these are generic. Is not worth adding them to `highLevelApi*` instead? (I get that if they're imported - the end user will just see `api` as the package.

File: pkg/api/hosttwin.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		return err
+	}
+
+	twinID := register.NewRegisteredIdentity(twinKeypair, twinIssuer)
+
+	return TwinDelegatesControlToAgent(resolverClient, twinID, agentID, delegationName)
+}
+
+// GetOwnershipOfTwinByPrivateExponent Get Ownership of a twin using the private exponent of the twin.
+func GetOwnershipOfTwinByPrivateExponent(resolverClient register.ResolverClient, twinIssuer *register.Issuer, twinPrivateExponent string, newOwnerID register.RegisteredIdentity, newOwnerKeyName string) error {
```
Comment (vtermanis) [inline pkg/api/hosttwin.go:23]: Is this *Set* (or *Add*) instead of *Get*, judging by what this does?

Comment (vtermanis) [reply pkg/api/hosttwin.go:23]: Oh ok, "Get ownership" means... "Take ownership"? But I think "Get" might make people think of getter methods. Would "Take" or "Add" (is it replace) work better maybe?

File: pkg/api/hosttwin.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"fmt"
+
+	"github.com/Iotic-Labs/iotics-identity-go/pkg/advancedapi"
+	"github.com/Iotic-Labs/iotics-identity-go/pkg/register"
+)
+
+// DelegateControlByPrivateExponent registers a twin identity with a control delegate to agent
+func DelegateControlByPrivateExponent(resolverClient register.ResolverClient, twinIssuer *register.Issuer, twinPrivateExponent string, agentID register.RegisteredIdentity, delegationName string) error {
```
Comment (vtermanis) [inline pkg/api/hosttwin.go:11]: Worth having *hex* either in the the input var or func name? (then would be in line with other funcs like `GetKeyPairFromPrivateExponentHex`) .. or would that be too wordy?

File: pkg/api/hosttwin.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	}
+
+	dProof, err := advancedapi.CreateProof(agentID.KeyPair(), agentID.Issuer(), []byte(twinIssuer.Did))
+	if err != nil {
+		return err
+	}
+
+	opts := []register.RegisterDocumentOpts{
+		register.AddFromExistingDocument(twinDoc),
+		register.AddPublicKey(newOwnerKeyName, agentID.KeyPair().PublicKeyBase58, false),
```
Comment (vtermanis) [inline pkg/api/hosttwin.go:53]: Is this not `newOwnerID.KeyPair().PublicKeyBase58` like in above func?

**Review by vtermanis — COMMENTED** [review]
Looking good - got only some minor Qs

File: pkg/api/hosttwin.go (hunk 3 above)
Comment (timchurchard) [reply pkg/api/hosttwin.go]: I've changed them from Get to Take and added Hex on the end.

File: pkg/api/hosttwin.go (hunk 2 above)
Comment (timchurchard) [reply pkg/api/hosttwin.go]: I've moved these to high level APi

File: pkg/api/hosttwin.go (hunk 5 above)
Comment (timchurchard) [reply pkg/api/hosttwin.go]: Thanks, fixed

**Review by vtermanis — APPROVED** [review]
LGTM from my perspective (regarding the comments I had) :+1:

**Review by StephanieBracaloni — APPROVED** [review]
LGTM
